src/common.cc

```diff
@@ -507,9 +507,10 @@ namespace sharp {
         }
       }
     }
+
     // Limit input images to a given number of pixels, where pixels = width * height
     if (descriptor->limitInputPixels > 0 &&
-      static_cast<uint64_t>(image.width() * image.height()) > descriptor->limitInputPixels) {
+      static_cast<uint64_t>(image.width()) * static_cast<uint64_t>(image.height()) > descriptor->limitInputPixels) {
```

Comment:
You could also cast only `image.width()` to `uint64_t`, `image.height()` is then implicitly converted to `uint64_t`. Then, the multiplication is still done with two operands of type `uint64_t`.
```suggestion
      static_cast<uint64_t>(image.width()) * image.height() > descriptor->limitInputPixels) {
```
(helps to avoid the 120 column width, which this line almost exceeds)

Reply:
Good to know :), Haven't written a line of C++ in 11 years. I'll push it in a bit.